### PR TITLE
Add custom Transformer block example with comprehensive tests

### PR DESCRIPTION
**Pull request by xingzihai**
## Summary

This PR adds a custom Transformer block example with comprehensive tests, demonstrating how to create backend-agnostic transformer components using Keras 3.

### New Files

1. **`examples/demo_custom_transformer_block.py`** - Custom transformer components:
   - `PositionalEncoding` layer implementing sinusoidal positional encodings from "Attention Is All You Need" (Vaswani et al., 2017)
   - `TransformerBlock` layer combining multi-head self-attention, feed-forward networks, and residual connections
   - `create_transformer_classifier()` function demonstrating model composition

2. **`examples/demo_custom_transformer_block_test.py`** - Comprehensive test suite:
   - **PositionalEncodingTest**: Tests for shape preservation, encoding formula, serialization, batch independence
   - **TransformerBlockTest**: Tests for shape preservation, configuration validation, training vs inference modes, residual connections
   - **TransformerClassifierTest**: Tests for model creation, compilation, training, inference
   - **PositionalEncodingCorrectnessTest**: Mathematical correctness tests for encoding formula
   - **TransformerBlockCorrectnessTest**: Tests for self-attention and feed-forward network structure
   - **IntegrationTest**: Full pipeline tests including gradient flow and reproducibility

## Key Features

- All layers use backend-agnostic Keras operations (`keras.ops`) for compatibility with TensorFlow, JAX, and PyTorch backends
- Follows Keras 3 testing conventions using `testing.TestCase` and `run_layer_test`
- Comprehensive test coverage including:
  - Shape preservation
  - Serialization and deserialization
  - Configuration validation
  - Training vs inference mode behavior
  - Mathematical correctness
  - Gradient flow

## Testing

Tests can be run with:
```bash
pytest keras/examples/demo_custom_transformer_block_test.py
```

Or with a specific backend:
```bash
KERAS_BACKEND=jax pytest keras/examples/demo_custom_transformer_block_test.py
```

## Checklist

- [x] Code follows Keras style guidelines
- [x] Docstrings follow Keras conventions
- [x] Tests cover all major functionality
- [x] Backend-agnostic operations used throughout

### CODE REVIEW

**Comment by hertschuh** [comment]
@xingzihai 

This type of example doesn't belong in this repository.
- transformer architectures are implemented in https://github.com/keras-team/keras-hub
- examples are added in https://github.com/keras-team/keras-io
